changemakers/frappe_changemakers/doctype/case/case.py

```diff
@@ -8,8 +8,23 @@
 class Case(Document):
 	def before_save(self):
 		self.set_created_by()
+		self.calculate_total_amount()
+		self.validate_total_amount()
 
 	def set_created_by(self):
 		if not self.created_by:
 			owner = frappe.db.get_value("User", self.owner, "full_name")
 			self.created_by = owner
+   
+	def calculate_total_amount(doc):
+		total_amount = 0
+		for row in doc.get("payment_details"):
+			total_amount += row.amount
+		doc.total_amount = total_amount
```

Comment:
Maybe try changing this code after adding the `Payment Details` child table field in `Case` Doctype?

Reply:
Making the changes @harshtandiya